File: backend/02_microservicios_producto/src/main/java/com/market/model/Producto.java
package com.market.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name="Productos")
public class Producto {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idProducto;
    private String nombre;
    private double precioUnitario;
    private int stock;
    private int idCategoria;
}

File: frontend/src/app/model/Producto.ts
export class Producto{
    idProducto:number;
    nombre:string;
    unidades:number;
    precioUnitario:number;
    stock:number;
    idCategoria:number;

    constructor()
    {
        this.idProducto=0;
        this.nombre="";
        this.unidades=0;
        this.precioUnitario=0;
        this.stock=0;
        this.idCategoria=0;
    }
}